### PR TITLE
DCOS-12703: Display correct service address across the UI

### PR DESCRIPTION
**Pull request by nLight**
This PR ensures we display a consistent and hopefully correct value for a service address across the UI.

To make it possible:
- VIP label generation function has been extracted
- Multi and Single container network forms have been changed
- Multi and Single container config review screens have been changed

Additionally I've made more strict the rule of checking if a VIP has been changed by a user, yet there's an outstanding issue I decided to extract https://mesosphere.atlassian.net/browse/DCOS-13394

### CODE REVIEW

**Comment by nLight** [comment]
wake up and
retest this please 🙏

File: plugins/services/src/js/utils/ServiceConfigUtil.js [hunk 1]
```diff
@@ -19,6 +19,25 @@ const ServiceConfigUtil = {
...
+  },
+
+  buildHostNameFromVipLabel(label, port) {
+    const [ipOrName, labelPort] = label.split(':');
+
+    // Sometimes we don't know the port yet
+    port = port || '<assigned port>';
+
+    // 0 means randomly assigned port
+    port = labelPort === '0' ? port : labelPort;
```
Comment (wavesoft) [inline plugins/services/src/js/utils/ServiceConfigUtil.js]: Do we need to consider the case where `labelPort` is undefined if `label` does not contain `:` ?

File: plugins/services/src/js/utils/__tests__/ServiceConfigUtil-test.js [hunk 2]
```diff
@@ -61,4 +61,31 @@ describe('ServiceConfigUtil', function () {
...
+      ).toEqual('192.168.0.1:<assigned port>');
+    });
+
+    it('correctly uses desired port', function () {
+      expect(
+        ServiceConfigUtil.buildHostNameFromVipLabel('192.168.0.1:0', 9070)
+      ).toEqual('192.168.0.1:9070');
+    });
+
+  });
```
Comment (wavesoft) [inline plugins/services/src/js/utils/__tests__/ServiceConfigUtil-test.js]: We can also add a test case for `buildHostNameFromVipLabel('192.168.0.1', 9070)`

**Comment by nLight** [comment]
retest this please

File: plugins/services/src/js/utils/ServiceConfigUtil.js (hunk 1 above)
Comment (nLight) [inline plugins/services/src/js/utils/ServiceConfigUtil.js]: Not sure if it's a valid value. Marathon runs the app with `VIP_0: 'some-app'` though, and doesn't change the label.

Comment (nLight) [inline plugins/services/src/js/utils/ServiceConfigUtil.js]: Extracted this corner case https://mesosphere.atlassian.net/browse/DCOS-13519
as it is a bit bigger

**Comment by nLight** [comment]
retest this please

**Comment by jfurrow** [comment]
run integration tests

**Comment by jfurrow** [comment]
@wavesoft Are you happy with this as it is? Just wanted to double check that your comments get addressed. This looks ready to me! :ship:

**Comment by wavesoft** [comment]
✨ :shipit:
